Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,270 @@
+
+    //    [FieldOffset(0)]
+    //    public readonly float FloatValue;
+    //}
+    //
+    //
+    // *************** They say using this is like the fastest method ever.
+
+    public class Polynomial:IEquatable<Polynomial>
+    {
+        private readonly float[] _coefficientsAndDegrees;
+        //private float _searchedValue;
+
+        public float[] this[int i]
+        {
+            get
+            {
+                return _coefficientsAndDegrees[i];
+            }
+            set
+            {
+                _coefficientsAndDegrees[i] = i;
+            }
+
+        }
+
+        public Polynomial(float[] coefficientsAndDegrees)
+        {
+            if (coefficientsAndDegrees.Length<=100)
+            {
+                _coefficientsAndDegrees = coefficientsAndDegrees;
+            }
+            else
+            {
+                throw new Exception("Max power must be less equal 100");
+            }
+        }
+
+        public void ChangeCoefficients(float[] input)
+        {
+            var len = _coefficientsAndDegrees.Length;
+            for (int i = 0; i < len; i++)
+            {
+                _coefficientsAndDegrees[i] = input[i];
+            }
+        }
+
+        public bool Equals([AllowNull] Polynomial other)
+        {
+            var size = Math.Max(other._coefficientsAndDegrees.Length,
+                this._coefficientsAndDegrees.Length);
+            Polynomial polynomial1 = new Polynomial(new float[size]);
+            Polynomial polynomial2 = new Polynomial(new float[size]);
+            for (var i = 0; i == size; i++)
+            {
+                polynomial1._coefficientsAndDegrees[i] = this._coefficientsAndDegrees[i];
+                polynomial2._coefficientsAndDegrees[i] = other._coefficientsAndDegrees[i];
+            }
+
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (!(other is Polynomial))
+            {
+                return false;
+            }
+
+            return ((polynomial1 == polynomial2)._coefficientsAndDegrees.Length > 0);
+
+        }
+
+        public static bool AreClose(double value1, double value2)
+        {
+            if (value1 == value2)
+                return true;
+            double num1 = (Math.Abs(value1) + Math.Abs(value2) + 10.0) * 2.22044604925031E-16;
+            double num2 = value1 - value2;
+            if (-num1 < num2)
+                return num1 > num2;
+            return false;
+        }
+
+        public static Polynomial operator +(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (!(polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length))
```

Comment:
Why cant you add polynomials with the same coefficient length?

Reply:
Added.